hera_cal/io.py

```diff
@@ -312,6 +323,12 @@ def __init__(self, input_data, filetype='uvh5', **read_kwargs):
         else:
             raise NotImplementedError('Filetype ' + self.filetype + ' has not been implemented.')
 
+        # save longest and shortest integration times in the file for later use in up/downsampling
+        # if available, these will be used instead of the ones in self.integration_time during partial I/O
+        if self.integration_time is not None:
+            self.longest_integration = np.max(self.integration_time)
+            self.shortest_integration = np.min(self.integration_time)
```

Comment:
should this be defined as None if integration_time is None?

Reply:
We could. I used `if hasattr(self, 'shortest_integration'):` instead. Do you have a strong preference?

Reply:
no strong preference, it just seems better from a bookkeeping perspective

Reply:
Done.